conn.go

```diff
@@ -283,40 +314,61 @@ func (l *Conn) processMessages() {
 		select {
 		case l.chanMessageID <- messageID:
 			messageID++
-		case messagePacket, ok := <-l.chanMessage:
+		case message, ok := <-l.chanMessage:
 			if !ok {
 				l.Debug.Printf("Shutting down - message channel is closed")
 				return
 			}
-			switch messagePacket.Op {
+			switch message.Op {
 			case MessageQuit:
 				l.Debug.Printf("Shutting down - quit message received")
 				return
 			case MessageRequest:
 				// Add to message list and write to network
-				l.Debug.Printf("Sending message %d", messagePacket.MessageID)
-				l.chanResults[messagePacket.MessageID] = messagePacket.Channel
-				// go routine
-				buf := messagePacket.Packet.Bytes()
+				l.Debug.Printf("Sending message %d", message.MessageID)
+				l.chanResults[message.MessageID] = message.Channel
+
+				if l.RequestTimeout != 0 {
+					go func() {
```

Comment:
rather than letting two things race to feed a single results channel, would it make sense to use a `select{}` block to select between a timeout and the message.Channel, and take the winner and feed *that* to the results channel?

```
if l.RequestTimeout != 0 {
  outWithTimeout := make(chan *PacketResponse)
  l.chanResults[message.MessageID] = outWithTimeout
  go func(){
    ... handle panic ...
    select {
      case <-time.After(l.RequestTimeout):
        outWithTimeout <- ... timeout error response
      case result := <-message.Channel:
        outWithTimeout <- result
    }
  }()
} else {
  l.chanResults[message.MessageID] = message.Channel
}
```